gunicorn_thrift/gevent_worker.py

```diff
@@ -45,6 +45,10 @@ def handle(self, listener, client, addr):
 
             try:
                 while True:
+                    self.nr += 1
+                    if self.alive and self.nr >= self.max_requests:
+                        self.log.info("Autorestarting worker after current process.")
+                        self.alive = False                    
```

Comment:
You should change `while True` to `while self.alive`, it can still process requests after the current process unless master murdered the process..